### PR TITLE
Add api change log for replacing report_card.dataset with report_card.type

### PR DESCRIPTION
**Pull request by qnkhuat**

### CODE REVIEW

File: docs/developers-guide/api-changelog.md [hunk 1]
```diff
@@ -5,3 +5,6 @@ title: API changelog
 # Breaking changes to the API interface
 
 ## Metabase 0.49.0
+- `POST /api/card` and `PUT /api/card/:id`
```
[the remark this answers is not in the document]
Comment (qnkhuat) [reply docs/developers-guide/api-changelog.md:8]: I think we need to only mentions the mutate endpoints, not sure if enumerating all the endpoints that return card give much benefit here.

File: docs/developers-guide/api-changelog.md [hunk 2]
```diff
@@ -5,3 +5,6 @@ title: API changelog
...
+
+  `dataset` is deprecated and will be removed in future version. Instead use `type=model` (equivalent to `dataset=true`) or `type=query` (equivalent to `dataset=false`).
```
[the remark this answers is not in the document]
Comment (qnkhuat) [reply docs/developers-guide/api-changelog.md:10]: that's better :)

File: docs/developers-guide/api-changelog.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (calherries) [reply docs/developers-guide/api-changelog.md:8]: I was thinking the warning about switch to `type` in the future applies to `GET /api/card`, even if it's not a breaking change. But fair enough this changelog is about breaking changes not warnings

**Review by calherries — APPROVED** [review]

**Comment by calherries** [comment]
I've removed the milestone so as to not add an entry in the release changelog

**Pull request merged**